### PR TITLE
Fixes EMNIST split and label issues

### PR DESCRIPTION
**Pull request by vballoli**
Fixes https://github.com/pytorch/vision/issues/2630

### CODE REVIEW

File: torchvision/datasets/mnist.py [hunk 1]
```diff
@@ -265,6 +265,9 @@ def __init__(self, root: str, split: str, **kwargs: Any) -> None:
         super(EMNIST, self).__init__(root, **kwargs)
         self.classes = self.classes_split_dict[self.split]
 
+        if self.target_transform is None and self.split == 'letters':
+            self.target_transform = lambda x: x - 1
```
Comment (fmassa) [inline torchvision/datasets/mnist.py:269]: To me I think it might be preferable to change `classes_split_dict` case for `letters`, if it's not compatible with the current annotations.

I think that implicitly changing in the `target_transform` the labels might be misleading to users, and would make previously trained models to not be valid anymore.

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

I've one comment, let me know what you think

File: torchvision/datasets/mnist.py (hunk 1 above)
Comment (vballoli) [reply torchvision/datasets/mnist.py:269]: I did think of that initially while fixing this, but the issue is the class to letter mapping is still offset by 1. The only other solution would be to add a dummy letter to position 0 to fix the class offset. For ex: 1 is A while loading the dataset, but classes_split_dict would return b. I added this fix since transforms is a cleaner approach, but adding a dummy letter to the list is preferred, I'm happy to change it to that too.

Comment (fmassa) [reply torchvision/datasets/mnist.py:269]: > add a dummy letter to position 0

I think this would be the preferred approach IMO. Maybe add something like `__unused__` ?

Comment (datumbox) [reply torchvision/datasets/mnist.py]: I picked up @vballoli's great work to make the final changes and merge.

@fmassa I used `N/A` instead because COCO uses this value for the unused categories.

File: torchvision/datasets/mnist.py [hunk 2]
```diff
@@ -249,13 +249,13 @@ class EMNIST(MNIST):
     md5 = "58c8d27c78d21e728a6bc7b3cc06412e"
     splits = ('byclass', 'bymerge', 'balanced', 'letters', 'digits', 'mnist')
     # Merged Classes assumes Same structure for both uppercase and lowercase version
-    _merged_classes = set(['C', 'I', 'J', 'K', 'L', 'M', 'O', 'P', 'S', 'U', 'V', 'W', 'X', 'Y', 'Z'])
-    _all_classes = set(list(string.digits + string.ascii_letters))
+    _merged_classes = {'c', 'i', 'j', 'k', 'l', 'm', 'o', 'p', 's', 'u', 'v', 'w', 'x', 'y', 'z'}
```
Comment (datumbox) [inline torchvision/datasets/mnist.py:252]: One remark concerning cosmetics, not actual functionality. The [EMNIST paper](https://arxiv.org/pdf/1702.05373v1.pdf) describes removing the capital letters. We can actually write the classes in a way that achieves this but then the labels will look weird; something like: `A, B, c, D, ...`. So I think @vballoli's solution is fine.

File: torchvision/datasets/mnist.py (hunk 1 above)
Comment (fmassa) [reply torchvision/datasets/mnist.py]: Using `N/A` is fine with me

**Review by fmassa — APPROVED** [review]
Thanks @vballoli and @datumbox !

File: torchvision/datasets/mnist.py (hunk 2 above)
Comment (fmassa) [reply torchvision/datasets/mnist.py:252]: Yeah, I think this is fine as is, but thanks for pointing it out
